qiskit_experiments/data_processing/nodes.py

```diff
@@ -481,6 +481,87 @@ def _population_error(self, counts_dict) -> Tuple[float, float]:
         return p_mean, np.sqrt(p_var)
 
 
+class DirichletProbability(Probability):
+    r"""Compute probability and variance from count dictionary.
+
+    This node is a subtype of :py:class:`~qiskit_experiments.data_processing.nodes.Probability`,
+    in which variance is computed based on a binomial distribution at the risk of zero variance
+    at probability at either zero or one.
+
+    This node avoid such singularity by assuming Dirichlet distribution with Bayes update
+    taking a prior distribution. Namely, a mean value is replaced by a mode that represents
+    the most likely value to be sampled
+
+    .. math::
+
+        p = \frac{\alpha_i - 1}{\alpha_0 + K}
+
+    where :math:`\alpha_i = f_i + \theta_i`, :math:`\alpha_0 = \sum_{i=1}^K \alpha_i`,
+    :math:`\theta_i` is the prior distribution and :math:`f_i` is
+    the count of :math:`i`-th element of :math:`K` dimensional vector.
```

Comment:
Is K the number of shots here?

Reply:
K is the number of possible element, i.e. if two qubits are measured, the possible states are 00, 01, 10, 11 and thus K=4.